test/integration/docker_exec_websocket_server_test.js

```diff
@@ -0,0 +1,92 @@
+var DockerWorker = require('../dockerworker');
+var TestWorker = require('../testworker');
+var assert = require('assert');
+var cmd = require('./helper/cmd');
+var debug = require('debug')('docker-worker:test:docker_exec_websocket_server_test')
+// var waitForEvent = require('../../lib/wait_for_event');
+var slugid = require('slugid');
+var DockerExecClient = require('docker-exec-websocket-server').DockerExecClient;
+var request = require('superagent-promise');
+var Promise = require('promise');
+
+suite('use docker exec websocket server', () => {
+  let worker;
+  setup(async () => {
+    worker = new TestWorker(DockerWorker);
+    await worker.launch();
+  });
+
+  teardown(async () => {
+    if (worker) {
+      await worker.terminate();
+      worker = null;
+    }
+  });
+  test('cat', async () => {
+    let taskId = slugid.v4();
+  	let task = {
+      payload: {
+        image: 'taskcluster/test-ubuntu',
+        command: cmd('sleep 40'),
+        maxRunTime: 2 * 60,
+        features: {
+          interactive: true
+        }
+      }
+    };
+    debug('posting to queue');
+
+    var client;
+
+    let resultPromise = worker.postToQueue(task, taskId);
+    var passed = false;
+    setTimeout(async () => {
```

Comment:
Instead of doing `setTimeout` you have two better options:
 1) `await base.testing.sleep(2000)` (basically a sleep utility)
 2) Use `await base.testing.poll(async () => { /* something that can fail */ })`

(2) is the most reliable one... It'll simply retry until the function given succeeds. Note that it does fail after some time and number of retries, this is configurable, see:
https://github.com/taskcluster/taskcluster-base/blob/master/testing.js#L529-L539

By default it'll try 16 times, and sleep 250ms between each retry... This is more robust than sleeping and then option docker boots quickly. Imagine a case where docker has to pull the image or something.